scripts/gis-material-update/validate-deploy/modules/common.py

```diff
@@ -0,0 +1,91 @@
+import math
+from sqlalchemy import create_engine, text
+    
+def deploy(pg_conn_uri, tormays_table_org, tormays_table_temp, logger=None):
+    # validate data amount: is it between given limits
+    delete_result = None
+    engine = create_engine(pg_conn_uri)
+    with engine.connect() as connection:
+        transaction = connection.begin()
+        delete_sql = "delete from " + tormays_table_org
+        try:
+            logger.info("Deploy ...")
+            delete_result = connection.execute(text(delete_sql))
+            logger.info(tormays_table_org + ": " + f"{delete_result.rowcount} rows deleted.")
+        except Exception as e:
+            # Transaction implicitly rolls back if an exception occurred within the "with" block
+            logger.error(f"Error: {e}")
+        
+        if delete_result:    
+            copy_sql = "insert into " + tormays_table_org + " select * from " + tormays_table_temp
+            try:
+                copy_result = connection.execute(text(copy_sql))
+                transaction.commit()
+                logger.info(tormays_table_org + ": " + f"{copy_result.rowcount} rows inserted.")
+            except Exception as e:
+                # Transaction implicitly rolls back if an exception occurred within the "with" block
+                logger.error(f"Error: {e}")
+            
+def validate_data_count_limits(module, pg_conn_uri, tormays_table_org, tormays_table_temp, validate_limit_min, validate_limit_max, logger=None):
+    # validate data amount: is it between given limits
+    tormays_result = None
+    temp_result = None
+    retval = None
+    old_amount = None
+    new_amount = None
+    tormays_sql = "select count(*) lkm from " + tormays_table_org + " where geometry is not null"
+    temp_sql = "select count(*) lkm from " + tormays_table_temp + " where geometry is not null"
```

Comment:
I tried running the code with hsl data, and I got the error:
```
psycopg2.errors.UndefinedTable: relation "tormays_buses_polys_temp" does not exist
```

When looking at the code I can't find anything that would create that table or anything that would insert data to it. `validate_data_count_limits` checks that there's data in the temp table. `deploy` copies the data from the temp table to the production table. But I can't see anything that would insert anything to the temp table. Am I missing something?

Reply:
I see, the temp table is created in the process stage. I changed my DB config to reference the same database I run the backend against and tried to run gis-validate-deploy again. That's why it didn't have the table in place.

This doesn't feel right in terms of responsibility separation. The process stage also does a part of the deployment. It feels like the processing should finish before the deployment starts. We're leveraging the functionality meant for writing debug data to a temporary table for writing the production data to the production database.

I think there should be a separate configuration for the production database, using the same `HAITATON_HOST`, `HAITATON_PORT`, `HAITATON_USER` and `HAITATON_PASSWORD` environment variables the cloud environments use.

Reply:
Could the temp table be created at the beginning of the validate-deploy stage from the output of the process stage? I.e. the `.gpkg` file, so in this case, `haitaton-gis-output/tormays_buses_polys.gpkg`.

Reply:
Now temp table is not created at all because data is coming from file.